packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -196,22 +196,53 @@ ${formatPluginOptionsForCLI(pluginConfig.getOriginalPluginOptions(), errors)}`,
   }
 
   let currentSyncData
-  const basicSyncConfig = {
-    limit: pageLimit,
-    resolveLinks: false,
-  }
+  let currentPageLimit = pageLimit
+  let lastCurrentPageLimit
+  let syncSuccess = false
   try {
     syncProgress = reporter.createProgress(
       `Contentful: ${syncToken ? `Sync changed items` : `Sync all items`}`,
-      pageLimit,
+      currentPageLimit,
       0
     )
     syncProgress.start()
-    reporter.verbose(`Contentful: Sync ${pageLimit} items per page.`)
-    const query = syncToken
-      ? { nextSyncToken: syncToken, ...basicSyncConfig }
-      : { initial: true, ...basicSyncConfig }
-    currentSyncData = await client.sync(query)
+    reporter.verbose(`Contentful: Sync ${currentPageLimit} items per page.`)
+
+    while (!syncSuccess) {
+      try {
+        const basicSyncConfig = {
+          limit: currentPageLimit,
+          resolveLinks: false,
+        }
+        const query = syncToken
+          ? { nextSyncToken: syncToken, ...basicSyncConfig }
+          : { initial: true, ...basicSyncConfig }
+        currentSyncData = await client.sync(query)
+        syncSuccess = true
+      } catch (e) {
+        // Back off page limit if responses content length exceeds Contentfuls limits.
+        if (
+          e.responseData.data.message.indexOf(`Response size too big`) !== -1 &&
```

Comment:
Includes is easier to read
```suggestion
          e.responseData.data.message.includes(`Response size too big`) &&
```